### ISSUE TITLE
AoT 1.1.16-7 TypeaheadMatch item is protected 

### ISSUE THREAD

**Issue by philjones88** [valor-software/ngx-bootstrap#1361]
I use this field to get the whole object but now trying to compile with AoT + 1.1.16-7 I get:

`Property 'item' is protected and only accessible within class 'TypeaheadMatch' and its subclasses.`

My code for example:

```
    public typeaheadOnSelect(e: TypeaheadMatch): void {
        this.logger.debug('typeaheadOnSelect', e);

        let selectedItem = e.item as ListItemViewModel; // My Class type

        this.members.push(selectedItem);

        this.member = '';
    }
```

**Comment by valorkin** [comment]
Can you do a small PR and make it public?

**Comment by philjones88** [comment]
@valorkin sure, I'll do it now. I have a workaround of using `any` to get past it, so not blocking :)

**Comment by quiringk** [comment]
@valorkin I am getting this error when building AOT

```
Error: Error at C:/Projects/NeilKellyClient4/aot/node_modules/ng2-bootstrap/timepicker/timepicker.component.ngfactory.ts:650:35: Supplied parameters do not match any signature of call target.
Error at C:/Projects/NeilKellyClient4/aot/node_modules/ng2-bootstrap/timepicker/timepicker.component.ngfactory.ts:668:35: Supplied parameters do not match any signature of call target.
    at check (C:\Projects\NeilKellyClient4\node_modules\@angular\tsc-wrapped\src\tsc.js:31:15)
    at Tsc.typeCheck (C:\Projects\NeilKellyClient4\node_modules\@angular\tsc-wrapped\src\tsc.js:86:9)
    at C:\Projects\NeilKellyClient4\node_modules\@angular\tsc-wrapped\src\main.js:33:23
    at process._tickCallback (internal/process/next_tick.js:103:7)
    at Module.runMain (module.js:606:11)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
Compilation failed
```

Is there something I need to do in order to make this work for timepicker? Here is what my timepicker template code looks like: (I tried both (ngModelChange) and (change) but both didn't work)

`<timepicker [(ngModel)]="date2" (change)="timeChanged()" [hourStep]="1" [minuteStep]="15"></timepicker>
`

**Comment by valorkin** [comment]
published as 1.1.16-8
docs preview here: http://valorkin.github.io/ng2-bootstrap/#/
